Some edits to READMEs (#57)
c4dt/arti-rest: README.md

# Arti-rest

> :warning: **Warning: arti-rest is not secure in all situations** Arti-rest modifies several core parts of `arti` and therefore does not provide the same guarantees that arti and the stock Tor client would. Security depends on whether these weakened guarantees are acceptable in your situation.

Arti-rest is a simple wrapper around [arti](https://gitlab.torproject.org/tpo/core/arti) that leverages arti to enable mobile apps to easily make anonymous REST requests via the Tor network. Arti-rest makes two changes on top of arti. First, arti-rest provides a simple wrapper for REST requests, to make calling HTTP(S) API endpoints easier. Second, arti-rest provides the option to use a customized, and therefore potentially much smaller, consensus. Using a customized consensus is less secure than using the original Tor consensus. It therefore depends on your application whether this reduction is worth while.

> Arti-rest is written in Rust and can therefore be compile into native libraries that is easy to integrate into mobile applications. All credits for enabling this approach go to the authors of [arti](https://gitlab.torproject.org/tpo/core/arti). Using a native library ensures that arti-rest can be bundled with Android and iOS applications, rather than relying on external Tor proxies that must be installed separately.

Reviewer: ```suggestion
Arti-rest is written in Rust and can therefore be compiled into native libraries that are easy to integrate into mobile applications. All credits for enabling this approach go to the authors of [arti](https://gitlab.torproject.org/tpo/core/arti). Using a native library ensures that arti-rest can be bundled with Android and iOS applications, rather than relying on external Tor proxies that must be installed separately.
```

# Arti-rest

Arti-rest is a simple wrapper around [arti](https://gitlab.torproject.org/tpo/core/arti) that leverages arti to enable mobile apps to easily make anonymous REST requests via the Tor network. Arti-rest makes two changes on top of arti. First, arti-rest provides a simple wrapper for REST requests, to make calling HTTP(S) API endpoints easier. Second, arti-rest provides the option to use a customized, and therefore potentially much smaller, consensus. 

> > :warning: **Warning: arti-rest is not secure in all situations** Arti-rest modifies several core parts of `arti`. These modifications result on arti-rest not providing the same guarantees than arti and the stock Tor client. Whether these weakened guarantees are acceptable in your situation. See the reliability section below to check what aspects of your system you need to consider to decide whether arti-rest is secure for you.

Reviewer 2: ```suggestion
> :warning: **Warning: arti-rest is not secure in all situations** Arti-rest modifies several core parts of `arti`. These modifications result on arti-rest not providing the same guarantees than arti and the stock Tor client. You will have to verify on your own whether these weakened guarantees are acceptable in your situation. See the reliability section below to check what aspects of your system you need to consider to decide whether arti-rest is secure for you.
```

       arti
```
       
`Arti-rest` provides a synchronous interface to send REST requests over tor, and
to receive the answers. It also does the TLS handling of the connection. The
asynchronous part needs to be done by the Android and iOS library.

## Reliability

Even though we tried our best to produce good code, this should not be used in a
mission critical piece of software without proper consideration of the following:

- arti itself mentions it's not ready for prime-time since it does not implement all protections that the stock Tor client does, see their 
  [README](https://gitlab.torproject.org/tpo/core/arti/-/blob/main/README.md). In particular, this might mean that you should not use Arti in production.

> - arti-rest uses a customized consensus to reduce bandwidth cost for infrequent use of arti-rest. As a result, the consensus is not signed by the default Tor directory authorities, but instead by the creator of the app. This approach in general is not secure. However, since the app developers already determines which code is loaded (and could therefore disable consensus validation altogether, or deanonymize users directly), it is likely acceptable in the case of mobile applications that directly include arti-rest.

Reviewer: ```suggestion
- arti-rest uses a customized consensus to reduce bandwidth cost for infrequent use of arti-rest. As a result, the consensus is not signed by the default Tor directory authorities, but instead by the creator of the app. This approach in general is not secure. However, since the app developer already determines which code is loaded (and could therefore disable consensus validation altogether, or deanonymize users directly), it is likely acceptable in the case of mobile applications that directly include arti-rest.
```
